### PR TITLE
[EarlyReturn] Enable ReturnBinaryAndToEarlyReturnRector in early-return config set

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Comment by TomasVotruba** [comment]
Thanks!

Oh, we don't have it enabled in `rector-ci.php`:
https://github.com/rectorphp/rector/blob/11d36a4bf99062cb5159eba29a013ef145b40a24/rector-ci.php#L59

Should be added there too. I forgot to handle that in the past.

There might be many false positives in this set, that will need to be fixed.

**Comment by samsonasik** [comment]
I've enable EARLY_RETURN in rector-ci.php

**Comment by samsonasik** [comment]
added handling to keep comment if exists.

File: rules/dead-code/src/Comparator/Parameter/ParameterDefaultsComparator.php [hunk 1]
```diff
@@ -53,8 +53,10 @@ private function isMutuallyExclusiveNull(ParameterReflection $parameterReflectio
         if ($parameterReflection->getDefaultValue() === null && $param->default !== null) {
             return true;
         }
-
-        return $parameterReflection->getDefaultValue() !== null && $param->default === null;
+        if (! ($parameterReflection->getDefaultValue() !== null)) {
```
Comment (samsonasik) [inline rules/dead-code/src/Comparator/Parameter/ParameterDefaultsComparator.php:56]: not identical need to be changed to identical and vice versa

Comment (samsonasik) [reply rules/dead-code/src/Comparator/Parameter/ParameterDefaultsComparator.php]: handled.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: packages/better-php-doc-parser/src/ValueObject/PhpDocNode/AbstractTagValueNode.php [hunk 2]
```diff
@@ -189,8 +189,10 @@ private function shouldPrintEmptyBrackets(): bool
         )) {
             return true;
         }
-
-        return $this->tagValueNodeConfiguration->hasOpeningBracket() && $this->tagValueNodeConfiguration->hasClosingBracket();
+        if (! $this->tagValueNodeConfiguration->hasOpeningBracket()) {
+            return false;
+        }
+        return (bool) $this->tagValueNodeConfiguration->hasClosingBracket();
```
Comment (TomasVotruba) [inline packages/better-php-doc-parser/src/ValueObject/PhpDocNode/AbstractTagValueNode.php:195]: `hasClosingBracket()` returns bool, so type-casting is not needed here

File: packages/node-type-resolver/src/NodeTypeCorrector/PregMatchTypeCorrector.php [hunk 3]
```diff
@@ -108,7 +108,10 @@ private function getVariableUsages(Variable $variable): array
         }
 
         return $this->betterNodeFinder->find((array) $scope->stmts, function (Node $node) use ($variable): bool {
-            return $node instanceof Variable && $node->name === $variable->name;
+            if (! $node instanceof Variable) {
+                return false;
+            }
+            return (bool) ($node->name === $variable->name);
```
Comment (TomasVotruba) [inline packages/node-type-resolver/src/NodeTypeCorrector/PregMatchTypeCorrector.php:114]: This also returns bool, so no type cast is needed

```suggestion
            return $node->name === $variable->name;
```

File: rules/coding-style/src/Rector/Class_/AddArrayDefaultToArrayPropertyRector.php [hunk 4]
```diff
@@ -176,8 +176,6 @@ private function clearNotNullBeforeCount(Class_ $class, array $propertyNames): v
             if (! $node instanceof BooleanAnd) {
                 return null;
             }
-
-            // $this->value !== null
```
Comment (staabm) [inline rules/coding-style/src/Rector/Class_/AddArrayDefaultToArrayPropertyRector.php:180]: Were this comments deleted manually or should the rector preserve them?

Comment (samsonasik) [reply rules/coding-style/src/Rector/Class_/AddArrayDefaultToArrayPropertyRector.php:180]: it seems by rectify command that delete commented code

File: packages/better-php-doc-parser/src/ValueObject/PhpDocNode/AbstractTagValueNode.php (hunk 2 above)
Comment (samsonasik) [reply packages/better-php-doc-parser/src/ValueObject/PhpDocNode/AbstractTagValueNode.php:195]: implemented.

File: packages/node-type-resolver/src/NodeTypeCorrector/PregMatchTypeCorrector.php (hunk 3 above)
Comment (samsonasik) [reply packages/node-type-resolver/src/NodeTypeCorrector/PregMatchTypeCorrector.php]: implemented.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/coding-style/src/Rector/Class_/AddArrayDefaultToArrayPropertyRector.php (hunk 4 above)
Comment (TomasVotruba) [reply rules/coding-style/src/Rector/Class_/AddArrayDefaultToArrayPropertyRector.php:180]: @samsonasik Could you return manually those comments? 

It's out of scope this pull-request, so just to preserve them here

Comment (samsonasik) [reply rules/coding-style/src/Rector/Class_/AddArrayDefaultToArrayPropertyRector.php]: reverted manually.

**Comment by TomasVotruba** [comment]
Thank you :clap:

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

**Pull request merged**